core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -785,4 +796,782 @@ private void writeAtomic(final String blobName, final BytesReference bytesRef) t
         }
     }
 
+    @Override
+    public void snapshot(IndexShard shard, SnapshotId snapshotId, IndexCommit snapshotIndexCommit, IndexShardSnapshotStatus snapshotStatus) {
```

Comment:
Yes, I did not modify anything, except taking the shard instead of looking it up using the IndicesService.